core/mm/core_mmu.c

```diff
@@ -2381,7 +2381,14 @@ paddr_t virt_to_phys(void *va)
 	return pa;
 }
 
-#if defined(CFG_TEE_CORE_DEBUG)
+/*
+ * Don't use check_va_matches_pa() for RISC-V, as its callee
+ * arch_va2pa_helper() will call it eventually, this creates
+ * indirect recursion and can lead to a stack overflow.
+ * Moreover, if arch_va2pa_helper() returns true, it implies
+ * the va2pa mapping is matched, no need to check again.
+ */
+#if defined(CFG_TEE_CORE_DEBUG) && !defined(RV64)
```

Comment:
Only for `RV64`, not for `RV32`? it seems the implementation of the same for these 2 configurations.